### DISCUSSION TITLE
Issue With Python Project and Excel

### DISCUSSION THREAD

**Discussion by jvodonnell** [community/community#60253] [Programming Help]
### Body

Hey Devs,

I've been working on a Python project to help with my day to day job. The purpose of the program is take the raw data from my worksheet and replace that date into a docx file. I have the code, the program runs without error but my letters arn't generating. I can't understand what I'm doing wrong. I'll list my code below for reference. All my files are in the same folder and the file names match. 
Any help would be stellar! Thanks in advance:

import openpyxl
from docx import Document
from docx.shared import Pt
from docx.enum.text import WD_ALIGN_PARAGRAPH
from datetime import date

# Load the Excel sheet
workbook = openpyxl.load_workbook('py_cores.xlsx')
sheet = workbook.active

# Load the letter template
template_path = 'Certified Letter.docx'

# Set font properties for the template
font_name = 'Calibri'
font_size = Pt(14)

# Iterate through the rows in the data sheet
for row in sheet.iter_rows(min_row=2, values_only=True):
    if len(row) < 6:
        continue
    
    name, rma, part_num, description, cus_id = row

    # Create a new document from the template
    document = Document(template_path)

    # Set font properties for the document
    for paragraph in document.paragraphs:
        run = paragraph.runs[0]
        run.font.name = font_name
        run.font.size = font_size

    # Replace placeholders in the document with data
    for paragraph in document.paragraphs:
        paragraph.text = paragraph.text.replace('{{cus_id}}', str(cus_id))
        paragraph.text = paragraph.text.replace('{{name}}', str(name))
        paragraph.text = paragraph.text.replace('{{part_num}}', str(part_num))
        paragraph.text = paragraph.text.replace('{{rma}}', str(rma))
        paragraph.text = paragraph.text.replace('{{description}}', str(description))
        paragraph.text = paragraph.text.replace('{{date}}', date.today().strftime('%B %d, %Y'))
        paragraph.alignment = WD_ALIGN_PARAGRAPH.LEFT

    # Save the certified letter
    filename = f'{cus_id}.docx'
    document.save(filename)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sohaibaumer** [comment]
This looks OK however, you can:

-	Verify that the Excel file ('py_cores.xlsx') is located in the same folder as your Python script. Double-check the filename and ensure it matches exactly.
-	Check that the 'Certified Letter.docx' template file is also in the same folder. Again, verify the filename and make sure it matches.
-	Add some print statements or use a debugger to check the values of variables like name, rma, part_num, description, and cus_id to ensure they contain the expected data.
-	Confirm that the placeholders in the document template ('Certified Letter.docx') match the ones in your code (e.g., '{{cus_id}}', '{{name}}', etc.). Make sure they are spelled correctly and have the same formatting.
-	Check if any exceptions are being raised during the execution of the code. Wrap the code inside a try-except block and print or log any exceptions that occur to get more detailed error information.
-	Ensure that you have the necessary libraries installed, such as openpyxl, docx, and datetime.
